builder/bundles.go

```diff
@@ -420,24 +419,35 @@ func resolvePackages(numWorkers int, set bundleSet, packagerCmd []string, emptyD
 	return &bundleRepoPkgs, err
 }
 
-func installFilesystem(chrootDir string, localPath string, packagerCmd []string, downloadRetries int) error {
+func installFilesystem(chrootDir string, packagerCmd []string, downloadRetries int, dnfDownloadDir string, repos map[string]repoInfo) error {
 	var rpmFull string
 	var err error
 
-	rpmFull = filepath.Join(localPath, fileSystemRpm)
-	rpmMap[fileSystemRpm] = true
+	rpm := fileSystemInfo.name + "-" + fileSystemInfo.version + "." + fileSystemInfo.arch + ".rpm"
+	rpmMap[rpm] = true
+
+	if repos[fileSystemInfo.repo].urlScheme == "file" {
+		rpmFull = filepath.Join(repos[fileSystemInfo.repo].url, rpm)
+	} else {
+		packagerCmdNew := merge(packagerCmd, "--destdir", dnfDownloadDir)
+		_, err := downloadRpms(packagerCmdNew, []string{fileSystemInfo.name}, chrootDir, downloadRetries)
+		if err != nil {
+			return err
+		}
+		rpmFull = filepath.Join(dnfDownloadDir, rpm)
+	}
+
 	if _, err = os.Stat(rpmFull); os.IsNotExist(err) {
-		if !Offline {
-			rpmPath, err := downloadRpm(packagerCmd, []string{"filesystem"}, chrootDir, downloadRetries)
-			if err != nil {
-				return err
-			}
-			rpmFull = filepath.Join(rpmPath, fileSystemRpm)
-			if _, err = os.Stat(rpmFull); err != nil {
-				return fmt.Errorf("rpm not found: %s", rpmFull)
-			}
-		} else {
-			return fmt.Errorf("rpm not found: %s", rpmFull)
+		// if rpm is not found, name of rpm file is not in autospec generated format. In this case
+		// we query the database to get the name of the rpm file containing the package.
+		rpm = queryRpmName(packagerCmd, fileSystemInfo.name, fileSystemInfo.repo)
```

Comment:
Similarly to my other comment, we should use NVR instead of fileSystemInfo.name to be more specific.